### PR TITLE
Support setting `k` and `threshold` in `python_filter_similarity`

### LINKED ISSUE #81: Python and C++ implementations of the filter similarity calculation are inconsistent
`python_filter_similarity` and `cffi_filter_similarity` differ in their results in various ways; one obvious example is that the Python version doesn't even accept the `k` and `threshold` arguments.

### PR DESCRIPTION
**Pull request by mhsiah**
Like it says. This means `python_filter_similarity` and `cffi_filter_similarity` now have the same interface and their outputs can be compared for all parameters. Everywhere that `python_filter_similarity` was called, `k = 1` and `threshold = 0.0` are passed, as these correspond to the previous behaviour of `python_filter_similarity`.

This PR incidentally adds some assertions to the `test_similarity.py` to tests that just ran code without checking results. It also uses more useful values of `k` and `threshold` in many of the tests (namely `default_k = 10` and `default_threshold = 0.5` instead of `k = 1` and `threshold = 0.0` as it had before).

Resolves #81.

### CODE REVIEW

File: anonlink/entitymatch.py [hunk 1]
```diff
@@ -3,31 +3,39 @@
...
 
 
-def python_filter_similarity(filters1, filters2):
+def python_filter_similarity(filters1, filters2, k, threshold):
     """Pure python method for determining Bloom Filter similarity
 
     Both arguments are 3-tuples - bitarray with bloom filter for record, index of record, bitcount
 
-    :return: A list of tuples *one* for each entity in filters1.
+    :return: A list of tuples *k* for each entity in filters1.
```
[the remark this answers is not in the document]
Comment (mhsiah) [reply anonlink/entitymatch.py]: I'm not against that proposal, but it would be nice to know what you think the advantage of passing `k=None` is over passing `k=len(whatever)`. Note that the latter is more explicit about what the intention is. Also note that, were we to allow `k=None`, the callee would require special code to check if `k==None` and, that being so, set `k=len(whatever)`, which is not obviously a net win to me.

In summary, I don't see a significant advantage to allowing `None`, and I think it obscures the intention a bit. (Note that I assume you're not proposing that `k=None` should be a default parameter. That could be a reasonable suggestion, but should happen over at #80.)
